flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlShowDatabasesConverter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.operations.converters;
+
+import org.apache.flink.sql.parser.dql.SqlShowDatabases;
+import org.apache.flink.table.catalog.CatalogManager;
+import org.apache.flink.table.operations.Operation;
+import org.apache.flink.table.operations.ShowDatabasesOperation;
+
+/** A converter for {@link SqlShowDatabases}. */
+public class SqlShowDatabasesConverter implements SqlNodeConverter<SqlShowDatabases> {
+
+    @Override
+    public Operation convertSqlNode(SqlShowDatabases sqlShowDatabases, ConvertContext context) {
+        if (sqlShowDatabases.getPreposition() == null) {
+            return new ShowDatabasesOperation(
+                    sqlShowDatabases.getLikeType(),
+                    sqlShowDatabases.getLikeSqlPattern(),
+                    sqlShowDatabases.isNotLike());
+        } else {
+            CatalogManager catalogManager = context.getCatalogManager();
+            String[] fullCatalogName = sqlShowDatabases.getCatalog();
+            String catalogName =
+                    fullCatalogName.length == 0
+                            ? catalogManager.getCurrentCatalog()
+                            : fullCatalogName[0];
```

Comment:
Do we need to get the current catalog at this point? You handle that during the execution already, don't you?

Reply:
Good point. 
As you pointed out, we already  get the `catalog` during the execution (`ShowDatabasesOperation::execute(Context ctx)`). We get the catalog from the `ctx` object.

However, we also use `catalogName` inside `ShowDatabasesOperation::asSummaryString()` that takes no arguments (i.e., no access to the `Context`). As a result,`asSummaryString` method will have no catalog name in its result.

Reply:
Won't it be more correct to not print the catalog in the summary string if it was not given by the user?

Reply:
That is also what we do. If no catalog is provided, summary string outputs no catalog name:

```
        final String sql1 = "SHOW DATABASES";
        assertShowDatabases(sql1, sql1);
```

The thing is if a user provides a catalog name, it does not necessarily execute the `ShowDatabaseOperation::execute(Context ctx)` method. For example, please see the below code snipped I used in tests. 

```
    private void assertShowDatabases(String sql, String expectedSummary) {
        Operation operation = parse(sql);
        assertThat(operation).isInstanceOf(ShowDatabasesOperation.class);
        final ShowDatabasesOperation showDatabasesOperation = (ShowDatabasesOperation) operation;
        assertThat(showDatabasesOperation.asSummaryString()).isEqualTo(expectedSummary);
    }
```


Moreover, we the catalog name provided in the constructor (`ShowDatabaseOperation`) is not necessarily the same as the one we obtain in `ShowDatabaseOperation::execute(Context ctx)`. That is why, we retrieve the catalog only if the `catalogName` is `null`:

```
        String cName =
                catalogName == null ? ctx.getCatalogManager().getCurrentCatalog() : catalogName;

```

Reply:
But, if the catalogName is null, doesn't it mean the user did not provide the name? What are the situation that `catalogName` can be null? Can it happen in any other situation than if there is no `FROM/IN catalog`?

To be honest the situation that the prepostion is not null, but the catalog name is empty looks like a bug to me. Is it an equivalent to `SHOW DATABASES FROM LIKE ...`? This doesn't look like a valid SQL statement.

Reply:
Sorry for the confusion. 
No, there is no case that `preposition` is `not null` and `catalog` is `null`. 
In fact, based on our discussion, I have already removed `preposition` field from `ShowDatabasesOperation`.

`catalog` is null (so the `preposition` that doesn't exist now) only when there is no `FROM/IN` clause. 

Do the changes in `ShowDatabasesOperation` reflect your comments or did I miss something?

Reply:
I am referring to the `SqlShowDatabasesConverter#convertSqlNode`:
```
        if (sqlShowDatabases.getPreposition() == null) {
...
        } else {
            String[] fullCatalogName = sqlShowDatabases.getCatalog();
            String catalogName =
                    fullCatalogName.length == 0
```

Is it possible that you have a preposition, but the catalog name is not present: `fullCatalogName.length == 0`. What is the equivalent SQL query that results in such a situation?

Reply:
ok, got it. Good point. No, if the preposition is not null, catalog name should be present. I updated `SqlShowDatabasesConverter `

Reply:
Great! Thank you!